src/community/schemaless-features/schemaless-core/src/main/java/org/geoserver/schemalessfeatures/filter/SchemalessPropertyAccessorFactory.java

```diff
@@ -101,9 +101,12 @@ private Object walkComplexAttribute(ComplexAttribute complexAttribute, String[]
                 if (result instanceof ComplexAttribute)
                     complexAttribute = (ComplexAttribute) result;
                 else if (result instanceof List) {
+                    boolean lastPart = (i + 1) == path.length;
                     @SuppressWarnings("unchecked")
                     List<Object> attributes = List.class.cast(result);
-                    List<Object> results = walkList(attributes, path, i);
+                    List<Object> results;
+                    if (!lastPart) results = walkList(attributes, path, i);
```

Comment:
Bit confused about this change... shouldn't it be in walkList? I see walkList/walkComplexAttribute recurse without ever doing this check, which leaves me wondering.

Reply:
you are right, moved it to walk list method.